modules/core/src/main/java/org/apache/fluo/core/oracle/OracleServer.java

```diff
@@ -412,17 +418,23 @@ public void takeLeadership(CuratorFramework curatorFramework) throws Exception {
       synchronized (this) {
         byte[] d = curatorFramework.getData().forPath(maxTsPath);
         currentTs = maxTs = LongUtil.fromByteArray(d);
+
+        isLeader = true;
       }
 
       gcTsTracker = new GcTimestampTracker();
       gcTsTracker.start();
 
-      isLeader = true;
-
       while (started) {
         // if leadership is lost, then curator will interrupt the thread that called this method
         Thread.sleep(100);
       }
+    } catch (IllegalStateException e) {
```

Comment:
The `LeaderSelector` appears to be swallowing this, rather than throwing it, so I don't think there's any catch we can do in our code to handle this.

Reply:
@keith-turner has an open PR in Curator if I am not mistaken?

Reply:
Would merging that and having the changes "percolate" here (not intended as a bad joke but hey) resolve this (even hypothetically?). If so I can try to help them out over there to make it happen faster if that is desirable.

Reply:
We are seeing lots of new exceptions after upgrading Curator.  This catch handled a case we were seeing with the older version of curator.  In the new version we are seeing exceptions within Curator and a ton of them.   I am looking into this trying to figure out if the new exceptions are a Curator or Fluo problem.

Reply:
I created #1037 in an attempt to solve the issues with spurious errors being logged.

Reply:
I opened [CURATOR-468](https://issues.apache.org/jira/browse/CURATOR-468) to report the problems I was seeing with leader selector.